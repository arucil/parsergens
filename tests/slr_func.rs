#![feature(proc_macro_hygiene)]

use insta::assert_debug_snapshot;

parsergens::parsergen!(mod parser_with_func: SLR("fixtures/expr_func.pg"));

#[test]
fn state() {
  let input = r"factorial(7) * 1000 + gcd(12, 39)";
  let result = parser_with_func::with_input(input)
    .parse_expr();
  assert_debug_snapshot!(result);
}