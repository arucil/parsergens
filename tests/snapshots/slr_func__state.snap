---
source: tests/slr_func.rs
expression: result
---
Ok(
    Ok(
        5040003.0,
    ),
)
